aws_lambda_powertools/utilities/parameters/ssm.py

```diff
@@ -20,7 +20,9 @@ class SSMProvider(BaseProvider):
     config: botocore.config.Config, optional
         Botocore configuration to pass during client initialization
     boto3_session : boto3.session.Session, optional
-            Boto3 session to use for AWS API communication
+            Boto3 session to use for AWS API communication, will not be used if boto3_client is not None
+    boto3_client: boto3.client, optional
```

Comment:
Let's use SSMClient from mypy boto3 project as it accurately defines what we're expecting, so mypy and other type checkers can catch potential issues later.

We can bring it as a dev dependency, and use `"SSMClient"` to benefit from Python's forward reference type annotation.

https://youtype.github.io/boto3_stubs_docs/mypy_boto3_ssm/client/

Reply:
fixed